router/proxy/metrics.go

```diff
@@ -0,0 +1,65 @@
+package proxy
+
+import (
+	"strconv"
+	"time"
+
+	router "github.com/flynn/flynn/router/types"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	prometheus.MustRegister(backendHTTPStatusMetric)
+	prometheus.MustRegister(backendHTTPConnErrorMetric)
+	prometheus.MustRegister(backendHTTPLatencyMetric)
+}
+
+var backendHTTPConnErrorMetric = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "router_backend_http_conn_errors_total",
+		Help: "Router backend HTTP connection errors",
+	},
+	[]string{"app", "service", "job_id"},
+)
+
+func trackBackendHTTPConnError(backend *router.Backend) {
+	backendHTTPConnErrorMetric.With(prometheus.Labels{
+		"app":     backend.App,
+		"service": backend.Service,
+		"job_id":  backend.JobID,
+	}).Inc()
+}
+
+var backendHTTPStatusMetric = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "router_backend_http_status_total",
+		Help: "Router backend HTTP status codes",
```

Comment:
This should be `router_backend_http_responses_total`, the status code is just another label.